lib/grammers-tl-types/src/lib.rs

```diff
@@ -84,19 +86,24 @@ pub use deserialize::{Cursor, Deserializable};
 pub use generated::{enums, functions, name_for_id, types, LAYER};
 pub use serialize::Serializable;
 
+#[cfg(feature = "impl-serde")]
+use serde_derive::{Deserialize, Serialize};
+
 /// This struct represents the concrete type of a vector, that is,
 /// `vector` as opposed to the type `Vector`. This bare type is less
 /// common, so instead of creating a enum for `Vector` wrapping `vector`
 /// as Rust's `Vec` (as we would do with auto-generated code),
 /// a new-type for `vector` is used instead.
+#[cfg_attr(feature = "impl-serde", derive(Serialize, Deserialize))]
 #[derive(Clone, Debug, PartialEq)]
 pub struct RawVec<T>(pub Vec<T>);
 
 /// This struct represents an unparsed blob, which should not be deserialized
 /// as a bytes string. Used by functions returning generic objects which pass
 /// the underlying result without any modification or interpretation.
+#[cfg_attr(feature = "impl-serde", derive(Serialize, Deserialize))]
 #[derive(Clone, Debug, PartialEq)]
-pub struct Blob(pub Vec<u8>);
+pub struct Blob(#[cfg_attr(feature = "impl-serde", serde(with = "serde_bytes"))] pub Vec<u8>);
```

Comment:
## Help needed. 

I am not formiliar with tl proto. Is there any other structure that store bytes in .tl definition files? All bytes-like properties need marked by `serde_bytes` for better performence.

Reply:
Yes, `Vec<u8>` are also generated for the TL `bytes` type. If that's the case, we may need to check the field types when generating code.

Might be good to write a `test` that makes sure this works as expected.

Reply:
Yes, `Vec<u8>` are also generated for the TL `bytes` type. If that's the case, we may need to check the field types when generating code.

Might be good to write a `test` that makes sure this works as expected.

Reply:
done in [b3573e3](https://github.com/Lonami/grammers/pull/277/commits/b3573e3d178636a25c2986519be1b5712fbebc3f)